### ISSUE TITLE
Breaking changes after update on 1.20.3 (urgently)

### ISSUE THREAD

**Issue by 7iomka** [greensock/GreenSock-JS#240]
Urgently help me please!
I developed animation for one of my sites.
After refactoring the code, I updated the version of the plugins, including the gsap and got a broken animation.
I read the update history and did not understand which of the properties is outdated.
Help me to understand!
1) Animation works as expected (v 1.19.1) https://codepen.io/7iomka/pen/ZaLGGZ?editors=1100
2) The animation is played to the end and it starts again abruptly (v 1.20.3) https://codepen.io/7iomka/pen/wPgawp?editors=1100
Maybe I built the loop incorrectly, but if so, why did it work before?
Thank you!

**Comment by jackdoyle** [comment]
I see a few problems:

1. GSDevTools requires 1.20.3.
2. GSDevTools doesn't do well with an infinitely repeating timeline. Since yours repeated infinitely, GSDevTools notices that and just shows the first iteration. (It's hard to scrub to Infinity, you know?)

So I'd just set your repeat to something like 100 so that it's not infinitely repeating (if your goal is to use GSDevTools with it). 

Does that help?

**Comment by 7iomka** [comment]
GSDevTools has nothing to do with it. I put it on my local site to experiment with the animation in the latest version, and in demonstrative in codepene I just copied it, but since you paid attention to it, I removed it from both demos.
Please look - refresh the links.

**Comment by jackdoyle** [comment]
Aha! Sorry about that. Misunderstood. Okay, here's a quote from the 1.20.0 release notes:

> "when adding an animation a timeline that contains an infinitely-repeating animation, instead of defaulting to adding it at the very end (which would be almost infinitely in the future), it'll use the most recently-added child's endTime as the default insertion point instead. Much more intuitive."

You've got several infinitely repeating child tweens/timelines. So for example, **after adding an infinitely repeating child**, you didn't define an insertion point for this label so it uses the default... 

```javascript
//in 1.19.1, this would be [almost] infinitely in the future! But in 1.20.0+, it'd be placed at the endTime() of the most recently added child.
baseTL.add('endLoop')
```

If you want the old behavior (placing it at the end of the timeline, even if it's almost infinitely in the future), you could do: 
```javascript
baseTL.add('endLoop', baseTL.duration());
```
So basically it looks like the problem is simply due to the difference in the default insertion times. 

Does that clear things up?

**Comment by 7iomka** [comment]
Thank you man!

**Comment by jackdoyle** [comment]
No problem at all. Happy tweening!
